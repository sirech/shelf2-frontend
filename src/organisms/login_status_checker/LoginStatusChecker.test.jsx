import React from 'react'

import { fullRender } from 'test'
import FullStatusChecker, { LoginStatusChecker } from './LoginStatusChecker'

describe('components', () => {
  describe('LoginStatusChecker', () => {
    let loginSuccess

    beforeEach(() => {
      loginSuccess = jest.fn()
      global.localStorage.getItem = jest.fn()
    })

    it('calls loginSuccess if the token is there', () => {
      global.localStorage = {
        getItem: (_) => 'token'
      }

      fullRender(<LoginStatusChecker loginSuccess={loginSuccess} />)
      expect(loginSuccess.mock.calls.length).toBe(1)
    })

    it('does not call loginSuccess if there is no token', () => {
      fullRender(<LoginStatusChecker loginSuccess={loginSuccess} />)
      expect(loginSuccess.mock.calls.length).toBe(0)
    })
  })

  describe('FullStatusChecker', () => {
    it('does not blow up', () => {
      const { component } = fullRender(<FullStatusChecker />)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
