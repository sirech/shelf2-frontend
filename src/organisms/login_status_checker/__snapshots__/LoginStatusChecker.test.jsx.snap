// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components FullStatusChecker does not blow up 1`] = `null`;
